2333/2/libs/top_bad_words_analyzer.rb

```diff
@@ -0,0 +1,26 @@
+require_relative 'handler'
+require 'terminal-table'
+require_relative './analyzer_printer.rb'
+require_relative './names_searcher.rb'
+
+# VersusBattle text analyzer
+class TopBadWordsAnalyzer
+  def initialize(rappers, take_value)
+    @rappers = rappers
+    @limit = take_value.to_i
+  end
+
+  def top_bad_words
+    AnalyzerPrinter.print_top_bad_words(top_sorted_rappers)
+  end
+
+  private
+
+  def top_sorted_rappers
+    sorted_rappers_by_number_of_bad_words.first(@limit)
+  end
+
+  def sorted_rappers_by_number_of_bad_words
+    @rappers.sort_by!(&:number_of_bad_words).reverse!
```

Comment:
Why do you mutate `@rappers` object here? There is no reason to do this and this may harm other parts of your app.
```rb
class Task
  def initialize(a)
    @a = a
  end

  def sort
    @a.sort!.reverse!
  end
end
```

```
irb(main):004:0> a = [3, 1, 6, 2, 7]
irb(main):014:0> a
=> [3, 1, 6, 2, 7]
irb(main):015:0> t = Task.new(a)
=> #<Task:0x00007f8f810646f0 @a=[3, 1, 6, 2, 7]>
irb(main):016:0> t.sort
=> [7, 6, 3, 2, 1]
irb(main):017:0> a
=> [7, 6, 3, 2, 1]
```


Rule of thumb: try to mutate as fewer things as possible. If you can avoid mutation — avoid it.